src/main/kotlin/com/qverkk/Main.kt

```diff
@@ -335,16 +338,14 @@ class Huawei(url: String, username: String = "admin", password: String?) {
         return this.checkResponse(response, mutator)
     }
 
-    fun doPostRequest(endpoint: String, data: Any,  mutator: Class<*>): Any? {
+    fun doPostRequest(endpoint: String, data: Login, mutator: Class<*>): Any? {
```

Comment:
Note: Umm this method should be universal so data should be some parent of Login that is common for other data models... also variable loginRequestBody shoud be something more generic like requestBody...

Reply:
It is set to Any in the newest version, i was tinkering around with it to see why it wasnt printing the xml values correctly :D

Reply:
Ah ok